### ISSUE TITLE
Citations from docstrings are not collected

### ISSUE THREAD

**Issue by jordifradera** [JuliaDocs/DocumenterCitations.jl#39]
I found no way to include a citation in a docstring of a method.

```julia
"""
     f(x)

A dummy function to show the problem [dummycitation](@cite).
"""
function f(x)
    x+1
end
```

Maybe there is one...

Citation is read from the references.bib, but not found later.
The following error show up:

```
Warning: Bibliography anchor for key 'dummycitation' not found.
└ @ DocumenterCitations .....\.julia\packages\DocumenterCitations\f0eDy\src\citations.jl:456
```

A **workaround** is to place the citation in a .md of your documentation pages.

`[dummycitation](@cite)`

Then the citation is rendered even inside doscstrings.

Using DocumenterCitations v1.2.0 with Julia 1,9.1.

**Comment by fredrikekre** [comment]
Just ran into this as well. I guess we don't traverse docstrings during the collection phase.

**Comment by goerz** [comment]
Haven’t had a chance yet to confirm, but I’ll make sure to fix this asap, later this week

**Comment by sjdaines** [comment]
And here... this used to work, now failing with Documenter.jl v1.0 and DocumenterCitations v1.2

Example of failure: https://github.com/PALEOtoolkit/PALEOcopse.jl/actions/runs/6207321855/job/16852451709

**Comment by goerz** [comment]
Indeed, this was a regression in version 1.2, where we don't recurse into docstring nodes during the collection step. Thus, any reference that was only cited in a docstring would be missed.

Should be fixed with #40

**Comment by jordifradera** [comment]
Many thanks!!
Awsome package that saves a huge amount of time.
